### PR TITLE
Added new shader decompiler

### PR DESCRIPTION
**Pull request by DHrpcs3**
DO NOT MERGE THIS PR

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -16,3 +16,9 @@
 [submodule "minidx9"]
 	path = minidx9
 	url = https://github.com/hrydgard/minidx9.git
+[submodule "common"]
+	path = common
+	url = https://github.com/RPCS3/common
+[submodule "rsx_program_decompiler"]
```
Comment (tambry) [inline .gitmodules]: Why make the shader recompiler a submodule? Wouldn't it make more sense to be part of the core?

Comment (DHrpcs3) [inline .gitmodules]: rsx_program_decompiler is usable tool for
1) test and fix shaders decompilation
2) extract shaders(/resources) from elf64
3) extract ucode from shader
much easier change code in one place than migrate changes between projects

Comment (tambry) [inline .gitmodules]: I see, thank you.

**Comment by DHrpcs3** [comment]
looks like appveyor is down

**Comment by DHrpcs3** [comment]
temporary appveyor unmarked as required

**Comment by vlj** [comment]
current master doesn't build

**Comment by vlj** [comment]
(6>LINK : fatal error LNK1181: cannot open input file 'rsx_decompiler.lib') apparently a missing dependency between projects

**Comment by paulsapps** [comment]
@vlj you pulled new submodule?

**Comment by tambry** [comment]
@paulsapps @vlj I had compiling issues due to "VCLibs Version: 14.0" or something like that missing. Found the library on some Turkish looking website.

**Comment by vlj** [comment]
yes I pulled the new submodule.

Issue is that the sln doesn't have dependencies between shader_code, rsx_common_compiler, and emucore. Building them manually works but I doubt appveyor will build them.

**Pull request merged**